src/sentry/models/monitor.py

```diff
@@ -18,11 +19,42 @@
     sane_repr,
 )
 
+SCHEDULE_FREQ_MAP = {
+    'yearly': rrule.YEARLY,
+    'monthly': rrule.MONTHLY,
+    'weekly': rrule.WEEKLY,
+    'daily': rrule.DAILY,
+    'hourly': rrule.HOURLY,
+    'minutely': rrule.MINUTELY,
+    'secondly': rrule.SECONDLY,
+}
+
 
 def generate_secret():
     return uuid4().hex + uuid4().hex
 
 
+def get_next_schedule(base_datetime, schedule_type, schedule):
+    if schedule_type == ScheduleType.CRONTAB:
+        itr = croniter(schedule, base_datetime)
+        next_schedule = itr.get_next(datetime)
+    elif schedule_type == ScheduleType.INTERVAL:
+        freq, interval = schedule
+        rule = rrule.rrule(
+            freq=SCHEDULE_FREQ_MAP[freq],
+            interval=interval,
+            dtstart=base_datetime,
+            count=2)
+        if rule[0] > base_datetime:
```

Comment:
kind of weird behavior but i suppose it makes sense - unsure if this is really how we'd approach it, but seems logical